drivers/power/ltc7841/ltc7841.h

```diff
@@ -0,0 +1,189 @@
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef __LTC7841_H__
+#define __LTC7841_H__
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include "no_os_i2c.h"
+#include "no_os_irq.h"
+#include "no_os_print_log.h"
+#include "no_os_delay.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "no_os_units.h"
+#include "no_os_alloc.h"
+
+/** X is set based on the pull configuration of the ADDR pin */
+#define TWO_BYTES_LENGTH            2
+#define LTC7841_ADDR                (0x20)
+#define REPEATED_START_MODE         1
+#define I2C_WRITE_LENGTH_FOR_READ   1
+#define I2C_DEFAULT_ARRAY_SIZE      2
+#define STOP_MODE                   0
+#define LTC7841_DEFAULT_ADDRESS_0   0x5A
+#define LTC7841_DEFAULT_ADDRESS_1   0x5B
+#define LTC7841_TOTAL_REGISTERS     0x16
+#define LTC7841_BYTE_LENGTH         1
+#define LTC7841_WORD_LENGTH         2
+#define LTC7841_I2C_WR_FRAME_SIZE   3
+/* this denotes the rsense scale of LTC7841 2(both channels) * 50 uV/ bit / 0.002 (rsense resistor), unit in uA*/
+#define CURRENT_SCALE               (20)
+/*this denotes the max voltage output of LTC7841 in millivolts*/
+#define LTC7841_MAX_VOUT            48000
+#define LTC7841_7_BIT_ADDRESS_MASK  0x7F
+#define LTC7841_OPERATION_OFF       0x00
+#define LTC7841_OPERATION_TURN_ON   0x80
+#define LTC7841_OPERATION_MARGIN_LOW 0x98
+#define LTC7841_OPERATION_MARGIN_HIGH 0xA8
+/*slew rate multiplier*/
+#define LTC7841_NOMINAL_SLEW_RATE   0b00
+#define LTC7841_SLOW_SLEW_RATE      0b01
+#define LTC7841_FAST_SLEW_RATE      0b11
+/*READ TEMPERATURE 1*/
+#define LTC7841_INTERNAL_DIE_TEMPERATURE 0b0
+#define LTC7841_TSNS_PIN_VOLTAGE         0b1
+#define READ_TEMPERATURE_1_BIT      2
+/*MFR MARGIN MAX VALUE*/
+#define MFR_MAX_VALUE               0x1FF
+/*MFR VOUT INITIALIZE TO HALF */
+#define MFR_VOUT_INIT              0x0FF
+#define WRITE_ONLY_REGISTERS_NUMBER 2
+#define ECOMM                       9
+/* specific only to DC2798A in ohms*/
+#define RSENSE_VALUE_IIN            (500)
+
+extern const uint8_t ltc7841_write_only_registers[WRITE_ONLY_REGISTERS_NUMBER];
+
+/* LTC7841 registers list */
+enum ltc7841_registers {
+	LTC7841_OPERATION = 1,
+	LTC7841_VOUT_MODE = 0x20,
+	LTC7841_STATUS_WORD = 0x79,
+	LTC7841_READ_VIN = 0x88,
+	LTC7841_READ_IIN = 0x89,
+	LTC7841_READ_VOUT = 0x8B,
+	LTC7841_READ_IOUT = 0x8C,
+	LTC7841_READ_TEMPERATURE_1 = 0x8D,
+	LTC7841_PMBUS_REVISION = 0x98,
+	LTC7841_MFR_IOUT_PEAK = 0xD7,
+	LTC7841_MFR_VOUT_PEAK = 0xDD,
+	LTC7841_MFR_VIN_PEAK = 0xDE,
+	LTC7841_MFR_TEMEPRATURE1_PEAK = 0xDF,
+	LTC7841_MFR_IIN_PEAK = 0xE1,
+	LTC7841_MFR_CLEAR_PEAKS = 0xE3,
+	LTC7841_MFR_VOUT_MARGIN_HIGH = 0xE5,
+	LTC7841_MFR_SPECIAL_ID = 0xE7,
+	LTC7841_MFR_VOUT_COMMAND = 0xE8,
+	LTC7841_MFR_CONFIG = 0xE9,
+	LTC7841_MFR_VOUT_MARGIN_LOW = 0xED,
+	LTC7841_MFR_RAIL_ADDRESS = 0xFA,
+	LTC7841_MFR_RESET = 0xFD
+};
```

Comment:
You may want to follow this: https://github.com/analogdevicesinc/no-OS/blob/a75c351f56ab2699d7b0fe8623ffcaf4e867f6ec/drivers/power/lt7182s/lt7182s.h#L50C1-L170C33

For consistency across PMBUS devices and other no-OS drivers.